### PR TITLE
chore(website): [playground] support enum values and remove compilerOptions filter

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] Addresses an existing open issue: related to #5121
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

further refine and improve visual config editor, this should correct potential issues with providing configs like in #5121

compilerOptions will no longer be filtered out from not supported values in visualEditor and enum values will be displayed correctly as dropdown fields

image

additionally fields, `moduleResolution, lib, plugins, typeRoots, paths, moduleDetection, baseUrl` will be ignored

related to changes done in #5112

### CODE REVIEW

File: packages/website/src/components/editor/config.ts [hunk 1]
```diff
@@ -13,6 +13,13 @@ export function createCompilerOptions(
     module: window.ts.ModuleKind.ESNext as number,
     ...tsConfig,
     jsx: jsx ? window.ts.JsxEmit.Preserve : window.ts.JsxEmit.None,
+    moduleResolution: undefined,
+    lib: undefined,
```
Comment (armano2) [inline packages/website/src/components/editor/config.ts:17]: this line can be removed after #5162 is merged

**Review by bradzacher — APPROVED** [review]
LGTM!

**Comment by bradzacher** [comment]
feel free to merge once you've resolved the conflict @armano2

**Comment by armano2** [comment]
@bradzacher done

**Comment by bradzacher** [comment]
Oh sorry I forgot you don't have admin rights to merge without a review.

**Pull request merged**